editor/src/components/assets.ts

```diff
@@ -248,6 +248,37 @@ export async function walkContentsTreeAsync(
   }
 }
 
+export function zipContentsTree(
+  firstTree: ProjectContentTreeRoot,
+  secondTree: ProjectContentTreeRoot,
+  onElement: (
+    fullPath: string,
+    firstContents: ProjectContentsTree | null,
+    secondContents: ProjectContentsTree | null,
+  ) => boolean,
+): void {
+  let keys: Set<string> = emptySet()
+  Object.keys(firstTree).forEach(keys.add, keys)
```

Comment:
this syntax is new to me – what happens if you don't include the second parameter `keys` here?

Reply:
From memory, it has all kinds of issues around `this`, because of reasons.